cmd/coder/main.go

```diff
@@ -15,6 +20,48 @@ import (
 func main() {
 	rand.Seed(time.Now().UnixMicro())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Add a custom SIGQUIT handler that outputs to stderr and a well-known file
+	// in the home directory. This also prevents SIGQUITs from killing the CLI.
+	go func() {
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGQUIT)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-sigs:
+			}
+
+			buf := make([]byte, 10_000_000)
+			stacklen := runtime.Stack(buf, true)
+
+			_, _ = fmt.Fprintf(os.Stderr, "SIGQUIT:\n%s\n", buf[:stacklen])
+
+			// Write to a well-known file.
+			dir, err := os.UserHomeDir()
+			if err != nil {
```

Comment:
Why write to home? I'd rather just have these files in my temp dir.

Reply:
This handler is now only applied to `server` and `agent`, and exits on SIGQUIT again